### PR TITLE
Ensure that the Issues key in JSON format is a list

### PR DESCRIPTION
**Pull request by rafikdraoui**
This makes the data more consistent and easier to use with other tools. For example, a third-party tool can iterate over `Issues` without having to make a preliminary null check.

Before:
```console
$ golangci-lint run --out-format=json | jq
{
  "Issues": null,
  "Report": {
    "Linters": [...]
  }
}
```

After:
```console
$ golangci-lint run --out-format=json | jq
{
  "Issues": [],
  "Report": {
    "Linters": [...]
  }
}
```

### CODE REVIEW

**Comment by SVilgelm** [comment]
hmm, cannot reproduce the `nolintnolint` issue on local

File: pkg/commands/run.go
Comment (SVilgelm) [inline pkg/commands/run.go:482]: this is weird, I'm not sure why `nolintnolint` fails on this.
@ldez do you have any ideas?

Comment (ldez) [reply pkg/commands/run.go]: It's a flaky behavior, increased/produced by the cache used by the golangci-lint GH action.
It's a real pain, I often have to relaunch the CI several times before I can merge a PR.

**Comment by ldez** [comment]
another way to set struct tags:
```go
type JSONResult struct {
	Issues []result.Issue `json:",omitempty"`
	Report *report.Data   `json:",omitempty"`
}
```

```console
$ golangci-lint run --out-format=json | jq
{
  "Report": {
    "Linters": [...]
  }
}
```

File: pkg/commands/run.go
Comment (ldez) [reply pkg/commands/run.go]: pb partially fixed in #2359

**Review by SVilgelm — APPROVED** [review]

**Comment by ldez** [comment]
@SVilgelm have you seen my suggestion?

**Comment by SVilgelm** [comment]
yeah, I saw that, but I think it is better to explicitly print an empty array instead of omitting empty values. In this case the tools that should parse an output can avoid the checks for null or if the value does not exist

**Comment by ldez** [comment]
I think it's better to use the same rule for all the fields.
`*report.Data` ->
https://github.com/golangci/golangci-lint/blob/aa3550ab837926f70329a449adb4d826517016b1/pkg/report/data.go#L3-L18

**Comment by rafikdraoui** [comment]
I agree with @Svilgelm that ensuring that there is always in the output a key named `Issues` whose value is an array would make it easier for parsing. I suspect that the list of issues is what external tools mostly care about, rather than the internal configuration of golangci-lint or its logging output in `Report`.

That said, if the decision is to not go forward with this, it does feel slightly better to have the key omitted rather than having it set to nil. But it doesn't change the fact that external tools will have to be aware of the quirk and work around it (which is fine, if the tool integrates with golangci-lint, then it already is committed to be "hard wired" to it :smiley:)

**Comment by SVilgelm** [comment]
We can improve the `Report`'s part In a separate PR, it requires some additional discussion

**Comment by SVilgelm** [comment]
I'm merging this one

**Comment by ldez** [comment]
@SVilgelm when we don't agree, maybe we can discuss a bit, and if we don't find an agreement maybe we invite another maintainer in the topic instead of just merging.

**Pull request merged**
